twine/utils.py

```diff
@@ -159,6 +160,16 @@ def normalize_repository_url(url: str) -> str:
     return urlunparse(parsed)
 
 
+def get_file_size(filepath: str) -> Tuple[float, str]:
+    """Return the file size and unit of size following PyPI's file size format."""
+    size_unit = "MB"
+    file_size = os.path.getsize(filepath) / (1024 * 1024)
+    if round(file_size, 1) == 0:
+        size_unit = "KB"
+        file_size = os.path.getsize(filepath) / 1024
+    return file_size, size_unit
```

Comment:
Thanks for breaking this out into a utility function. Two nitpicks: I'd use `filename` instead of `filepath` (because the former is more common in the code), and I'm not sure `PyPI's file size format` is as useful as a more explicit description.

Also, I think it could be simplified, e.g.:

```python
    file_size = os.path.getsize(filename) / 1024
    size_unit = "KB"

    if file_size > 1024:
        file_size = file_size / 1024
        size_unit = "MB"

    return file_size, size_unit
```

That said, I think it would be preferable for this to return a string, e.g.:

```python
    return f"{file_size:.1f} {size_unit}"
```

That would simplify the usage in `upload`, and give us a reusable utility for getting a consistently-formatted file size.